pkg/network/server.go

```diff
@@ -331,12 +334,33 @@ func (s *Server) startProtocol(p Peer) {
 			err = p.WriteMsg(m)
 		case <-timer.C:
 			// Try to sync in headers and block with the peer if his block height is higher then ours.
-			if p.Version().StartHeight > s.chain.BlockHeight() {
+			if p.LastBlockIndex() > s.chain.BlockHeight() {
 				err = s.requestBlocks(p)
 			}
 			if err == nil {
 				timer.Reset(s.ProtoTickInterval)
 			}
+			if s.chain.HeaderHeight() >= p.LastBlockIndex() {
+				block, err := s.chain.GetBlock(s.chain.CurrentBlockHash())
+				if err != nil {
+					return
```

Comment:
This breaks the server because you're quitting the protocol handler without closing the connection.